dvc/ignore.py

```diff
@@ -134,6 +140,8 @@ def isexec(self, path):
         return self.tree.isexec(path)
```

Comment:
Also, we've discussed this before, but all of these methods don't take `dvcignore` into account, which is wrong.